### ISSUE TITLE
Persistent storage breaks test modules

### ISSUE THREAD

**Issue by lasers** [ultrabug/py3status#1261]
```
$ python3 clock.py
Exception in `test_module` post_config_hook() (KeyError) storage.py line 20.
Traceback
KeyError: 'i3status_config_path'
  File "/py3status/py3status/module.py", line 133, in prepare_module
    self.module_class.post_config_hook()
  File "clock.py", line 171, in post_config_hook
    saved_format = self.py3.storage_get('time_format')
  File "/py3status/py3status/py3.py", line 963, in storage_get
    self._storage_init()
  File "/py3status/py3status/py3.py", line 945, in _storage_init
    self._storage.init(self._module._py3_wrapper, self._is_python_2)
  File "/py3status/py3status/storage.py", line 20, in init
    py3_wrapper.config['i3status_config_path']

terminating module test_module
```
EDIT: Could use `def kill()` to set the storage? Fewer writes to the data.

**Comment by lasers** [comment]
Thoughts on using `def kill()` to set the storage? Much less writes to the data.

**Comment by tobes** [comment]
This shows yet again why the tests need to run through the full stack rather than out of stack.  Maybe I prioritize this.

>  Could use def kill() to set the storage?

We discussed this.  I do not like the fact that I could write data and it doesn't hit the disk.  I think we are planning to move to batched writes so that we limit the number of writes to say once per x seconds.

It was decided that initially we keep things simple.  Currently there is minimal writing to storage as only clock uses it when we change the displayed format or timezone.

**Comment by lasers** [comment]
>move to batched writes so that we limit the number of writes to say once per x seconds.

Sounds good. Storage support for `stopwatch`, `pomodoro`, `timer` are on my list. I wonder if you want me to deprecate `rate_counter`'s thing too in #1026 ?

**Comment by tobes** [comment]
> rate_counter

would be good but is complicated because.

* We need to keep the old data
* The data is really important

The module is designed for billing clients so we need to make sure that data is not lost as it could have financial repercussions for a user

I think timer/stopwatch maybe should merge togeather.

pomodoro I think needs some big love and maybe a total rewrite but I think we should add some of the features it needs as common code first.

**Comment by lasers** [comment]
Closed via #1263.
